src/utils/use-onboarding-flow.ts

```diff
@@ -133,97 +66,28 @@ export const useOnboardingFlow = (
 
     const routes: PartialRoute<
       Route<keyof RootStackParamList, object | undefined>
     >[] = [initialOnboardingRoute];
     if (
-      !shouldShowUserCreationOnboarding &&
-      initialOnboardingScreen?.screenName
+      !nextOnboardingSection?.shouldShowBeforeUserCreated &&
+      initialOnboardingScreen?.name
     ) {
       routes.unshift({name: defaultInitialRouteName});
     }
     return {routes};
   };
 
   return {
-    nextOnboardingScreen,
-    getNextOnboardingScreen,
+    nextOnboardingSection,
+    getNextOnboardingSection,
     getInitialNavigationContainerState,
   };
 };
-
-const useShouldShowExtendedOnboarding = () => {
-  const {enable_extended_onboarding} = useRemoteConfig();
-  const {extendedOnboardingOnboarded} = useAppState();
-
-  const {userCreationOnboarded} = useAppState();
-  return (
-    enable_extended_onboarding &&
-    !extendedOnboardingOnboarded &&
-    !userCreationOnboarded // this is for backward compatibility
-  );
-};
-
-const useShouldShowUserCreationOnboarding = () => {
-  const {userCreationOnboarded} = useAppState();
-  return !userCreationOnboarded;
-};
-
-const useShouldShowTravelTokenOnboarding = () => {
-  const {authenticationType} = useAuthState();
-  const {mobileTokenOnboarded, mobileTokenWithoutTravelcardOnboarded} =
-    useAppState();
-  const {disable_travelcard} = useRemoteConfig();
-  const {tokens, mobileTokenStatus} = useMobileTokenContextState();
-  const inspectableToken = tokens.find((token) => token.isInspectable);
-  return (
-    !!inspectableToken &&
-    !inspectableToken?.isThisDevice &&
-    mobileTokenStatus === 'success' &&
-    authenticationType === 'phone' &&
-    ((!mobileTokenOnboarded && !disable_travelcard) ||
-      (!mobileTokenWithoutTravelcardOnboarded && disable_travelcard))
-  );
-};
-
-const useShouldShowLocationOnboarding = () => {
-  const {locationWhenInUsePermissionOnboarded} = useAppState();
-  const {status: locationWhenInUsePermissionStatus} = useGeolocationState();
-
-  return (
-    !locationWhenInUsePermissionOnboarded &&
-    locationWhenInUsePermissionStatus === 'denied'
-  );
-};
-
-const useShouldShowNotificationPermissionScreen = () => {
-  const {notificationPermissionOnboarded} = useAppState();
-
-  const validFareContracts = useValidRightNowFareContract();
-
-  const hasFareContractWithActivatedNotification =
-    useHasFareContractWithActivatedNotification();
-
-  const pushNotificationsEnabled = usePushNotificationsEnabled();
-  const {permissionStatus: pushNotificationPermissionStatus} =
-    useNotifications();
-
-  const pushNotificationPermissionsNotGranted =
-    Platform.OS === 'ios'
-      ? pushNotificationPermissionStatus === 'undetermined'
-      : pushNotificationPermissionStatus === 'denied';
-
-  return (
-    !notificationPermissionOnboarded &&
-    pushNotificationsEnabled &&
-    pushNotificationPermissionsNotGranted &&
-    validFareContracts.length > 0 &&
-    hasFareContractWithActivatedNotification
-  );
```

Comment:
Note: these useShouldShow hooks have been moved to `use-onboarding-sections.ts`, and the prefix is instead useCustomShouldShow. Also the onboarded state from `AppContext` (such as e.g. `notificationPermissionOnboarded`) is no longer needed as conditions inside these hooks, since that is now checked in `getNextOnboardingSection` anyway